### ISSUE TITLE
It is not working for network setting page?

### ISSUE THREAD

**Issue by faiyazalam** [tareq1988/wordpress-settings-api-class#97]
I created a network setting page but it is not working.
When I submit it is showing 404 page.
To fix this I just remove the action name "options.php" from the form.
Now it is not saving data.

**Comment by dmhendricks** [comment]
Yes, I encountered the same issue. The class is [hard-coding](https://github.com/tareq1988/wordpress-settings-api-class/blob/master/src/class.settings-api.php#L531) `options.php`.

I tried changing line 531 in `class.settings-api.php` to:

```
<form method="post" action="<?php echo $_SERVER[ 'REQUEST_URI' ]; ?>">
```

...which I believe to be correct, but the settings are still not saving. I have to look at it more closely.

**Comment by dmhendricks** [comment]
Well, including the above change, I got it to work with a **_total hack_**. Given [this example](https://github.com/tareq1988/wordpress-settings-api-class/blob/master/example/oop-example.php)...

I added something like this to manually update options when the page is submitted:
```
function __construct() {
  // ... other code ...

  // Update network options when form submit is detected
  if( isset( $_POST['option_page'] ) && $_POST['option_page'] == 'wedevs_basics' ) $this->save_network_options();
}

private function save_network_options() {
    if( $_POST[ 'wedevs_basics' ] ) {
      $options_array = [];
      foreach( $_POST[ 'wedevs_basics' ] as $option_name => $option_value ) {
        $options_array[ $option_name ] = $option_value;
      }
      update_option( 'wedevs_basics', $options_array );
    }
  }
}
```

**NB!** This is _not_ ideal and the code above may contain errors. It is heavily modified from my own code, which has a bunch of extra logic and tabs. I included it in case you are desperate. Caveat emptor.

I'm not sure how to retrieve a value from `wp_options` from a sub-site since it always tries to get it from the site's own wp_options. Does anyone know? get_option() and the multisite variants didn't work for me. We can't save to wp_sitemeta for get_network_option() or whatever, else the settings page doesn't populate. So, you'll probably have to create another hack...
```
// Retrieve value from wp_options
private function get_wpsac_network_option( $section_id, $option_name ) {
  // Note: We can't use $wpdb->prefix because it will append the blog ID, so we'll have to get the prefix from wp-config.php

  global $wpdb, $table_prefix;
  $result = $wpdb->get_col( $wpdb->prepare( "SELECT option_value FROM {$table_prefix}options WHERE option_name = '%s'", $section_id ) );

  if( !isset( $result[0] ) || !is_serialized( $result[0] ) ) return '';
  $result = unserialize( $result[0] );

  return isset( $result[ $option_name ] ) ? $result[ $option_name ] : '';

}
```

Again, not tested for errors... There is a better way, I just don't feel like finding it right now.

Good luck! Haha

**Comment by faiyazalam** [comment]
@dmhendricks 
To save time, I have not used this plugin for network admin setting page. I am using this only for admin setting page.
Hope the author will fix this soon. 

Thanks for your response.

**Comment by dmhendricks** [comment]
@faiyazalam - [Carbon Fields](https://carbonfields.net/) is about the only custom fields framework I've observed that supports [network settings pages](https://github.com/htmlburger/carbon-fields-docs/blob/master/documentation/10-containers/31-network.md). It is available as a [plugin](https://carbonfields.net/release-archive/) and also [via Composer](https://carbonfields.net/docs/carbon-fields-quickstart/).
